include/xgboost/tree_model.h

```diff
@@ -613,6 +626,107 @@ inline void RegTree::CalculateContributions(const RegTree::FVec& feat, unsigned
   out_contribs[split_index] += leaf_value - node_value;
 }
 
+// used by _compute_weight
+inline unsigned _seq_prod(unsigned a, unsigned b) {
+  unsigned out = 1.0;
+  for (unsigned i = a; i <= b; ++i) out *= i;
+  return out;
+}
+
+inline bst_float _compute_weight(unsigned num_fixed, unsigned num_integrated) {
```

Comment:
Interesting... Then the computation of Shapley weights might be not adding much "insult to injury" for shallow trees. But when I tried replacing this function with a simple silly calculation like `1/(num_fixed + num_integrated + 1)` I saw the time cut in half for a RF model, so I though there could be some potential here. Using lgamma alone would most likely make things worse.